cleanlab/object_detection/filter.py

```diff
@@ -0,0 +1,211 @@
+
+    _assert_valid_inputs(
+        labels=labels,
+        predictions=predictions,
+        method=scoring_method,
+    )
+
+    is_issue = _find_label_issues(
+        labels,
+        predictions,
+        scoring_method=scoring_method,
+        return_indices_ranked_by_score=return_indices_ranked_by_score,
+    )
+
+    return is_issue
+
+
+def _find_label_issues(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    *,
+    return_indices_ranked_by_score: bool = True,
+    scoring_method: str = "objectlab",
+):
+    """Internal function to find label issues based on passed in method."""
+
+    if scoring_method == "objectlab":
+        auxiliary_inputs = _get_valid_inputs_for_compute_scores(ALPHA, labels, predictions)
+
+        overlooked_scores_per_box = _compute_overlooked_box_scores(
+            alpha=ALPHA,
+            high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+            auxiliary_inputs=auxiliary_inputs,
+        )
+        overlooked_issues_per_box = _find_label_issues_per_box(
+            overlooked_scores_per_box, OVERLOOKED_THRESHOLD
+        )
+        overlooked_issues_per_image = _pool_box_scores_per_image(overlooked_issues_per_box)
+
+        badloc_scores_per_box = _compute_badloc_box_scores(
+            alpha=ALPHA,
+            low_probability_threshold=LOW_PROBABILITY_THRESHOLD,
+            auxiliary_inputs=auxiliary_inputs,
+        )
+        badloc_issues_per_box = _find_label_issues_per_box(badloc_scores_per_box, BADLOC_THRESHOLD)
+        badloc_issues_per_image = _pool_box_scores_per_image(badloc_issues_per_box)
+
+        swap_scores_per_box = _compute_swap_box_scores(
+            alpha=ALPHA,
+            high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+            auxiliary_inputs=auxiliary_inputs,
+        )
+        swap_issues_per_box = _find_label_issues_per_box(swap_scores_per_box, SWAP_THRESHOLD)
+        swap_issues_per_image = _pool_box_scores_per_image(swap_issues_per_box)
+
+        issues_per_image = (
+            overlooked_issues_per_image + badloc_issues_per_image + swap_issues_per_image
+        )
+        is_issue = issues_per_image > 0
+    else:
+        is_issue = np.full(
+            shape=[
+                len(labels),
+            ],
+            fill_value=-1,
+        )
+
+    if return_indices_ranked_by_score:
+        scores = get_label_quality_scores(labels, predictions)
+        sorted_scores_idx = issues_from_scores(scores, threshold=1.0)
+        is_issue_idx = np.where(is_issue == True)[0]
+        sorted_issue_mask = np.in1d(sorted_scores_idx, is_issue_idx, assume_unique=True)
+        issue_idx = sorted_scores_idx[sorted_issue_mask]
+        return issue_idx
+    else:
+        return is_issue
+
+
+def _find_label_issues_per_box(
+    scores_per_box: List[np.ndarray], threshold: float
+) -> List[np.ndarray]:
+    """Takes in a list of size ``N`` where each index is an array of scores for each bounding box in the `n-th` example
```

Comment:
don't let this delay the PR, but I would try to clean up the documentation of these helper methods that we eventually intend to make user-facing methods. That will make your life much easier in the future, while you still remember all the details of these methods!